vmm/src/vm_config.rs

```diff
@@ -1056,4 +1065,45 @@ impl VmConfig {
             self.cpus.max_vcpus
         }
     }
+
+    pub fn consume_fds(&mut self, fds: Vec<i32>) -> Result<(), PayloadConfigError> {
+        // Update self.external_fds, if appropriate.
+        let Some(external_fds) = self.external_fds.as_mut() else {
+            if fds.is_empty() {
+                return Ok(());
+            } else {
+                return Err(PayloadConfigError::FdCountMismatch);
+            }
+        };
+
+        if external_fds.ids.len() != fds.len() {
+            return Err(PayloadConfigError::FdCountMismatch);
+        }
+
+        if fds.is_empty() {
+            return Ok(());
+        }
+
+        external_fds.fds = fds;
+
+        // Consume net fds.
+        let Some(nets) = self.net.as_mut() else {
+            return Ok(());
+        };
+
+        for net in nets {
+            let Some(net_id) = net.id.clone() else {
+                continue;
+            };
+            let mut fds = vec![];
+            for idx in 0..external_fds.ids.len() {
```

Comment:
this is a very C-style way to iterate. I think this can be replaced with `for (id, idx) in 0..external_fds.ids.iter().enumerate()`